### PR TITLE
Software Requirement Specification

### PR DESCRIPTION
**Pull request by AlexMi-Ha**

### CODE REVIEW

File: documents/requirements/SRS.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+- Algebra (with Matrices)
+- Complex Number Arithmetic
+
+#### Streak System
+Here we want the player to get attached to the game more by introducing a streak system. The player gets a streak wich tracks the amount of consecutive days they (uninterupted) played. So if a player doesn't play for 24 hours, the streak gets reset.
+
+#### Stats tracking
+We want to track some statistics of the player to display in a Dashboard and let the Person see their progress.
+#### Level System
+This will introduce a leveling System where players get levels by simply playing wich will increase the difficulty over time.
```
Comment (Annalytic-programming) [inline documents/requirements/SRS.md:59]: "which" instead of "wich"

File: documents/requirements/SRS.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+- Dashboard - Stats
+- Game
+- Level
+- Streak
+For both:
+- How to play
+- Landing page
+
+#### Communication Interfaces
+The Frontend will communicate with the Backend with the HTTP-Protocol
```
Comment (Annalytic-programming) [inline documents/requirements/SRS.md:105]: http is 'hyper-text-transfer-protocol'. So is it hyper-text-transfer-protocol-protocol?

**Review by Annalytic-programming — COMMENTED** [review]
only small changes suggested

File: documents/requirements/SRS.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+The technology used will be:
+- Backend: ASP.Net Web API, Sqlite Database
+- Frontend: Angular Web app
+- IDE: Visual Studio 2022, Visual Studio Code
+- Project Management: Jira, Github, Discord
+
+## 3. Specific Requirements
+### 3.1 Functionality
+#### Accounts (Login, Registering, Profiles)
+We want to let the user create their own account to track their progress, play games and get notified if the would lose their streak.
```
Comment (jas20202) [inline documents/requirements/SRS.md:44]: "if they would lose their streak"

File: documents/requirements/SRS.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+#### Game loop
+This is the most important feature of our app. Here we will implement the main Excercises in a playful way. Players get a random Math problem from a specific category and can then try to solve it to get points and progress in the game.<br>
+Currently planned categories are:
+- Mental Arithmetic
+- Analysis (with Differentiation and Integration)
+- Algebra (with Matrices)
+- Complex Number Arithmetic
+
+#### Streak System
+Here we want the player to get attached to the game more by introducing a streak system. The player gets a streak wich tracks the amount of consecutive days they (uninterupted) played. So if a player doesn't play for 24 hours, the streak gets reset.
```
Comment (jas20202) [inline documents/requirements/SRS.md]: Change "So if a player doesn't play for 24 hours" to " So if a player doesn't play till midnight"

File: documents/requirements/SRS.md [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+
+#### Stats tracking
+We want to track some statistics of the player to display in a Dashboard and let the Person see their progress.
```
Comment (jas20202) [inline documents/requirements/SRS.md:57]: "dashbord" klein

File: documents/requirements/SRS.md (hunk 2 above)
Comment (AlexMi-Ha) [reply documents/requirements/SRS.md:105]: yes get gud

File: documents/requirements/SRS.md (hunk 1 above)
Comment (jas20202) [reply documents/requirements/SRS.md]: System auch klein bitte

**Review by jas20202 — CHANGES_REQUESTED** [review]

File: documents/requirements/SRS.md (hunk 5 above)
Comment (AlexMi-Ha) [reply documents/requirements/SRS.md]: fixed

File: documents/requirements/SRS.md (hunk 4 above)
Comment (AlexMi-Ha) [reply documents/requirements/SRS.md]: fixed

File: documents/requirements/SRS.md [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+- Dashboard - Stats
+- Game
+- Level
+- Streak
+For both:
+- How to play
+- Landing page
+
+#### Communication Interfaces
+The Frontend will communicate with the Backend with the HTTP-Protocol
```
Comment (Annalytic-programming) [inline documents/requirements/SRS.md:105]: avoid duplication
replace with with via
sounds cleaner

File: documents/requirements/SRS.md [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+The User Interface should be designed in an obvious way that the user can directly start learning without having to learn the Interface first. For further explanation we will have a "How-To" page covering how to play and the overall infrastructure of the site and we will have small tutorials in each excercise, helping the player when stuck.
+
+### 3.3 Reliability
+The Server should only be down for a very short period of time when a new version of the web app is deployed. All other downtimes have to be handled as quickly as possible.
+
+### 3.4 Performance
+#### Capacity
+The Backend should be able to handle many players at the same time. At no point should there be a waiting queue for the user.
+#### Storage
+All of the persistant storage will be handled by the Backend, so storage won't be a problem
```
Comment (jas20202) [inline documents/requirements/SRS.md:97]: add . at the end

**Review by jas20202 — CHANGES_REQUESTED** [review]

**Comment by AlexMi-Ha** [comment]
Dont merge till use cases are finished

File: documents/requirements/SRS.md (hunk 6 above)
Comment (AlexMi-Ha) [reply documents/requirements/SRS.md]: fixed

File: documents/requirements/SRS.md (hunk 7 above)
Comment (AlexMi-Ha) [reply documents/requirements/SRS.md]: fixed

File: documents/requirements/SRS.md [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+#### Response Time
+As this will be a kind of game, the perfomance - e.g. checking the answer - should be as fast as possible.
+
+### 3.5 Supportability
+#### Coding Standards
+We will use the most common clean code standards. To ensure a clean structure we will use a variation of [Clean Architecture by Jason Taylor](https://github.com/jasontaylordev/CleanArchitecture) for our C# API Backend.
+#### Testing Strategy
+The application will have a high test coverage to ensure it is working correctly
```
[the remark this answers is not in the document]
Comment (AlexMi-Ha) [reply documents/requirements/SRS.md]: fixed

**Review by Annalytic-programming — APPROVED** [review]
seems okay so far

**Review by Annalytic-programming — APPROVED** [review]

**Pull request merged**